src/resources/filters/main.lua

```diff
@@ -426,7 +426,9 @@ table.insert(quarto_filter_list, { name = "pre-render", filter = {} }) -- entry
 tappend(quarto_filter_list, quarto_layout_filters)
 tappend(quarto_filter_list, quarto_post_filters)
 table.insert(quarto_filter_list, { name = "post-render", filter = {} }) -- entry point for user filters
+table.insert(quarto_filter_list, { name = "pre-finalize", filter = {} }) -- entry point for user filters
```

Comment:
I wonder if we should use a slightly "less public" name here. For example, you can use `finalize-combined` directly, since that's the name of the first finalize filter.

We haven't documented this part yet, but the `pre-` and `post-` names are meant to be "stable API entrypoints". `pre-finalize` and `post-finalize` are good names, but I was planning on not advertising the "finalize" part of the pipeline, because I'm not sure it's really necessary. This isn't a big deal either way, and we can keep this pair of names without advertising them.

Reply:
I added `pre-finalize` just to follow the other names, as I needed the `post-finalize`. 

My understanding is that `post-render()` would have the same place anyway right. 

> For example, you can use finalize-combined directly, since that's the name of the first finalize filter.

I used the internal names to debug this - that is how I found `finalize-coalesce-raw` was the problem. 

I am ok to remove this and use `finalize-combined` for testing at start of finalize, but for end, I added `post-finalize` because I thought `finalize-typst-state` as last step would not be stable. 

> I wonder if we should use a slightly "less public" name here.

Ok with that totally. Though usually, undocumented means often not that public :) So if we don't document then it stays hidden (a little). Though we could make clear that it is not to be used except for testing. 

So overall: 
* Ok to remove `pre-finalize` 
* but `post-finalize` should stay right ? Or can we insert lua for testing at the very end another way ?

Reply:
> So overall:
> 
> * Ok to remove `pre-finalize`
> * but `post-finalize` should stay right ? Or can we insert lua for testing at the very end another way ?

You're right that we currently don't have a "at-the-very-end" entry point. I guess that's enough to convince me to keep `finalize` as a real "stage" of our filter. Let's just leave both `pre-finalize` and `post-finalize`. We already have redundant names anyway, and that's on purpose (it's in case we move the filter steps, then they might stop being redundant)